pyanaconda/modules/subscription/subscription.py

```diff
@@ -242,5 +441,438 @@ def set_system_purpose_with_task(self, sysroot):
         :return: a DBus path of an installation task
         """
         task = SystemPurposeConfigurationTask(sysroot, self.role, self.sla, self.usage, self.addons)
+        # set system purpose as applied once the task finishes running
+        task.stopped_signal.connect(lambda: self._set_is_system_purpose_applied(True))
         path = self.publish_task(SUBSCRIPTION.namespace, task)
         return path
+
+    # subscription
+
+    @property
+    def registered(self):
```

Comment:
Does the property describe the state of the installation environment or does it describe the state of the DBus module? The question applies also for `entitlement_attached` and `is_system_purpose_applied`. During the backport, we might need to reconsider the design of this module if the properties describe its state.

Reply:
Both - the DBus module mirrors the RHSM API & system state in this regard.

Reply:
The signal has been added & redundant setters dropped.